File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int cnt =0,res =0;
        nums.push_back(0);
        for(int i=0;i<nums.size();i++){
            if(nums[i]) cnt++;
            else{
                res=max(res,cnt); 
                cnt=0;
                while(i<nums.size() && nums[i]==0) i++; 
                i--; 
            }
        }
        return res;
    }
};

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        res,cnt=0,0 
        for i in nums: 
            if not(i): 
                cnt=0
            else: 
                cnt+=1 
            res=max([res,cnt])
        return res